### ISSUE TITLE
Installing dependencies via setuptools fails

### ISSUE THREAD

**Issue by clarkbreyman-yammer** [Azure/azure-sdk-for-python#728]
While the pip install method works fine, pip and easy_install are different enough that dependencies are not installed correctly. The symptom is that azure.mgmt and its sub-packages never get created. 

To reproduce, setup a virtual environment

```
> mkdir test-setup
> cd setup
> virtualenv venv
> source venv/bin/activate
```

then create file setup.py as:

```
#!/usr/bin/env python3
from setuptools import setup

setup(
    name='test-setup',
    version='0.0.dev0',
    packages=['test_setup'],
    install_requires=[
        'azure==2.0.0rc5'
    ]
)
```

Installing as `pip install -e .` works fine. If you destroy and recreate the virtualenv and try:
`./setup.py develop` or ``./setup.py install`. 

Test via `python -c import azure.mgmt`

**Comment by clarkbreyman-yammer** [comment]
Also discovered I can't build a wheel via pip then install from that wheel. Makes it hard to build libraries that pull in the azure-sdk transitively.

**Comment by lmazuel** [comment]
Thanks for the feedback! I will take a look asap.

**Comment by lmazuel** [comment]
After looking, I remember now that the setuptools install/develop is using egg install and not wheel (i.e. it downloads the source from PyPI and build an egg). We don't support egg install on the `azure` SDK for Python.
So, you have to use pip and wheel. I might override the "bdist_egg" command in our setup.py to fail explicitly and write a message to explain.

**Comment by lmazuel** [comment]
Finally found a solution in Python 3 with the help of @brettcannon , thanks to [PEP420](https://www.python.org/dev/peps/pep-0420/). Will do a fix asap.

Otherwise, there is no way to make it work in Python 2.7 with eggs. I will maybe add an error on "bdist_egg" command if Python 2.7 is used.

**Comment by lmazuel** [comment]
This is fixed in azure 2.0.0 for Python 2.7 and 3.x :)
Thanks!
